### PR TITLE
Fixes from multi-platform work

### PR DESCRIPTION
**Pull request by jhol**
Hi, I've tested pion on Mint, Debian Testing & Stable, Fedora 20 & 17, and MingW64 on 32-bit and 64-bit Windows 7, and these are the fixes I made.

I've also fixed all the warnings, and set `-Werror` so that they stay fixed.

### CODE REVIEW

**Comment by mikedickey** [comment]
Thanks for your patch. If you haven't already, please sign Splunk's Contribution License at http://www.splunk.com/goto/contributions

**Comment by jhol** [comment]
@mikedickey wow - you responded.

Sadly we've now removed pion from our project due to the lack of responsiveness, bugs and missing features.

Also, we rather object to the requirement to sign over our copyright in the CLA as a matter of principle. It's a pretty sucky thing to ask. However, in this case we give permission for you to have these patches under the CLA.

It's no longer a necessity for us, but it's a shame to waste good patches.

**Comment by mikedickey** [comment]
Sorry, github stopped sending email notifications for pull requests, and I'm trying to figure out why.
Just to be clear, you will need to approve the CLA via the web form I linked to before I can review these patches. It's company policy for all contributors to sign a CLA and aside from that, this is necessary to protect the integrity of the project anyway. We're starting to see open source projects getting shut down completely by nefarious "contributors" because the owners didn't take this precaution.

**Comment by jhol** [comment]
@mikedickey I have signed the CLA now.

File: build/pion-boost.inc [hunk 1]
```diff
@@ -134,25 +134,22 @@ if test "x$enable_tests" == "xno"; then
 	# Display notice if unit tests are disabled
 	AC_MSG_NOTICE([Unit tests are disabled])
 else
-	BOOST_TRY_LIB=unit_test_framework
-	BOOST_TRY_LINK="$BOOST_HOME_DIR/lib/libboost_${BOOST_TRY_LIB}${BOOST_LIB_EXTENSION}.a"
 	# check if static version of lib is available
-	if !(test -r "$BOOST_TRY_LINK"); then
```
Comment (mikedickey) [inline build/pion-boost.inc]: Why did you remove this conditional code? The dynamic version of the unit test library is not always available, and the current code does the same thing if it can't find the static version.

Comment (jhol) [inline build/pion-boost.inc]: @mikedickey Well I'm not sure if it's the right solution - but the issue is that on Debian you get both the static `.a` and the dynamic `.so` installed, but this code will prefer to link the `.a` file which causes problems for applications that link dynamically boost elsewhere.

Comment (mikedickey) [inline build/pion-boost.inc]: OK I think what I will do then (making the change as part of my merge branch) is keep the conditional logic, but make it prefer dynamic instead of static..

File: build/pion-setup.inc [hunk 2]
```diff
@@ -40,9 +40,9 @@ case "$build_os" in
 	# -pthread is implied (automatically set by compiler)
 	CPPFLAGS="$CPPFLAGS -D_REENTRANT"
 	#LDFLAGS="$LDFLAGS"
-	#PION_OPT_FLAGS="-O3 -ggdb -fomit-frame-pointer -fstrict-aliasing -fno-tree-pre -falign-loops -Wall -Wno-strict-aliasing -DNDEBUG"
-	PION_OPT_FLAGS="-O3 -ggdb -fomit-frame-pointer -fstrict-aliasing -Wall -Wno-strict-aliasing -DNDEBUG"
-	PION_DEBUG_FLAGS="-O0 -ggdb -Wall -Wno-strict-aliasing -fkeep-inline-functions"
+	#PION_OPT_FLAGS="-O3 -ggdb -fomit-frame-pointer -fstrict-aliasing -fno-tree-pre -falign-loops -Wall -Wextra -Werror -Wno-strict-aliasing -DNDEBUG"
+	PION_OPT_FLAGS="-O3 -ggdb -fomit-frame-pointer -fstrict-aliasing -Wall -Wextra -Werror -Wno-strict-aliasing -DNDEBUG"
+	PION_DEBUG_FLAGS="-O0 -ggdb -Wall -Wextra -Werror -Wno-strict-aliasing -fkeep-inline-functions"
```
Comment (mikedickey) [inline build/pion-setup.inc]: Unfortunately in this case, it's in the boost library not pion. I found another case like this, but I was able to work around both using #pragma so I didn't have to remove -Werror
